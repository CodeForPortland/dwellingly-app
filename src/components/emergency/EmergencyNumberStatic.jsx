import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../icon/Icon';

const EmergencyNumberStatic = (props) => {
  const { emergency } = props;
  const parsedPhoneNumberOne = props.parseNumber(emergency.phoneNumberOne);
  const parsedPhoneNumberTwo = props.parseNumber(emergency.phoneNumberTwo);
  return (
    <div className="emergencyNumberItem" key={`row-${emergency.id}`}>
      <div className="emergencyTitle">
        {!props.doneEditing ? emergency.contact : props.newContact}
        {emergency.phoneNumberThree !== null && emergency.phoneNumberThree !== ''
          && (
          <div className="emergencyNumberDescription">
            <p>
              {!props.doneEditing ? emergency.phoneNumberThree : props.newNumberThree}
            </p>
          </div>
          )
        }
      </div>
      <div className="emergencyNumberContainer">
        <div className="emergencyNumber">
        {parsedPhoneNumberOne.textContentOfNumber !== undefined ?
          <div>
            <a href={`tel:${!props.doneEditing ? emergency.phoneNumberOne : props.newNumberOne}`}>
            {!props.doneEditing ? parsedPhoneNumberOne.number : props.newNumberOne}
            </a>
            <span> {parsedPhoneNumberOne.textContentOfNumber}</span>
          </div> :
          <a href={`tel:${!props.doneEditing ? emergency.phoneNumberOne : props.newNumberOne}`}>
          {!props.doneEditing ? emergency.phoneNumberOne : props.newNumberOne}
          </a>
        }
        </div>
        {emergency.phoneNumberTwo !== null && emergency.phoneNumberTwo !== ''
          && (
          <div className="emergencyNumber">
            {parsedPhoneNumberTwo.textContentOfNumber !== undefined ?
              <div>
                <span>
                  |&nbsp;&nbsp;&nbsp;
                  {parsedPhoneNumberTwo.textContentOfNumber}:&nbsp;
                </span>
                <a href={`tel:${!props.doneEditing ? emergency.phoneNumberTwo : props.newNumberTwo}`}>
                  {!props.doneEditing ? parsedPhoneNumberTwo.number : props.newNumberTwo}
                </a>
              </div> :
              <div>
                |&nbsp;&nbsp;&nbsp;
                <a href={`tel:${!props.doneEditing ? emergency.phoneNumberTwo : props.newNumberTwo}`}>
                  {!props.doneEditing ? emergency.phoneNumberTwo : props.newNumberTwo}
                </a>
              </div>
            }
          </div>
          )
        }
        <div className="emergencyNumberEditButtons">
          <div className="emergencyNumberEditIcon" onClick={props.onEditingNumber} role="presentation">
            <Icon icon="pencil" />
          </div>
          <div className="emergencyNumberIcon" onClick={props.onArchivingNumber} role="presentation">
            <Icon icon="archive" />
          </div>
        </div>
      </div>
    </div>
  );
}

EmergencyNumberStatic.propTypes = {
  doneEditing: PropTypes.bool.isRequired,
  newContact: PropTypes.string.isRequired,
  newNumberOne: PropTypes.string.isRequired,
  newNumberTwo: PropTypes.string,
  newNumberThree: PropTypes.string,
  onEditingNumber: PropTypes.func.isRequired,
  onArchivingNumber: PropTypes.func.isRequired,
  parseNumber: PropTypes.func.isRequired,
  emergency: PropTypes.shape({
    contact: PropTypes.string.isRequired,
    phoneNumberOne: PropTypes.string.isRequired,
    phoneNumberTwo: PropTypes.string,
    phoneNumberThree: PropTypes.string
  }),
}

EmergencyNumberStatic.defaultProps = {
  emergency: null,
  newNumberTwo: "",
  newNumberThree: ""
}

export default EmergencyNumberStatic
