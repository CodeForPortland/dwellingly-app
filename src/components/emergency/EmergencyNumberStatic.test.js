import React from 'react';
import renderer from 'react-test-renderer';
import EmergencyNumberStatic from './EmergencyNumberStatic';

describe('EmergencyNumberStatic Component', () => {

  it('renders phoneNumberOne if it exists', () => {
    const props = {
      emergency: {
        phoneNumberOne: '',
      },
      }
    const EmergencyNumberStaticComponent = renderer.create(<EmergencyNumberStatic {...props}/>).toJSON();
    expect(EmergencyNumberStaticComponent).toMatchSnapshot();
  });

  it('renders phoneNumberTwo if it exists', () => {
    const props = {
      emergency: {
        phoneNumberOne: '',
        phoneNumberTwo: '',
      },
      }
    const EmergencyNumberStaticComponent = renderer.create(<EmergencyNumberStatic {...props}/>).toJSON();
    expect(EmergencyNumberStaticComponent).toMatchSnapshot();
  });
  it('renders phoneNumberThree if it exists', () => {
    const props = {
      emergency: {
        phoneNumberOne: '',
        phoneNumberTwo: '',
        phoneNumberThree: '',
      },
      }
    const EmergencyNumberStaticComponent = renderer.create(<EmergencyNumberStatic {...props}/>).toJSON();
    expect(EmergencyNumberStaticComponent).toMatchSnapshot();
  });
});
