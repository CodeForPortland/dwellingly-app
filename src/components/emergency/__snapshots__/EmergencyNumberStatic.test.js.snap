// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmergencyNumberStatic Component renders phoneNumberOne if it exists 1`] = `
<div
  className="emergencyNumberItem"
>
  <div
    className="emergencyTitle"
  >
    <div
      className="emergencyNumberDescription"
    >
      <p />
    </div>
  </div>
  <div
    className="emergencyNumberContainer"
  >
    <div
      className="emergencyNumber"
    >
      <div>
        <span>
          : 
        </span>
        <a
          href="tel:undefined"
        />
      </div>
    </div>
    <div
      className="emergencyNumberEditButtons"
    >
      <div
        className="emergencyNumberEditIcon"
        role="presentation"
      >
        <span
          className="icon icon--pencil"
        >
          <svg
            viewBox="0 0 50 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
            />
          </svg>
        </span>
      </div>
      <div
        className="emergencyNumberIcon"
        role="presentation"
      >
        <span
          className="icon icon--archive"
        >
          <svg
            viewBox="0 0 50 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M37.2 18h-25c-.8 0-1.5-.7-1.5-1.5s.7-1.5 1.5-1.5h24.9c.8 0 1.5.7 1.5 1.5S38 18 37.2 18zm-2.7-6.4H14.9c-.8 0-1.5-.7-1.5-1.5s.7-1.5 1.5-1.5h19.6c.8 0 1.5.7 1.5 1.5s-.7 1.5-1.5 1.5z"
            />
            <path
              d="M41.9 3.7a3.4 3.4 0 0 0-3.2-2.1H11.3c-1.4 0-2.6.8-3.2 2.1L0 23.4v23.8c0 1.1.9 2 2 2h46a2 2 0 0 0 2-2V23.4L41.9 3.7zM10.8 4.9c.1-.2.3-.3.4-.3h27.5c.2 0 .4.1.4.3l7.6 18.5H3.2l7.6-18.5zM36 35c-.9 1.1-4.4 4.7-10 5h-.8a15 15 0 0 1-10.9-5.1c-.5-.6-.4-1.4.1-1.9.6-.5 1.4-.4 1.9.1a12 12 0 0 0 9.4 4.1c4.5-.3 7.3-3.2 8.1-4.1.5-.6 1.4-.6 1.9-.2.7.7.8 1.5.3 2.1z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmergencyNumberStatic Component renders phoneNumberThree if it exists 1`] = `
<div
  className="emergencyNumberItem"
>
  <div
    className="emergencyTitle"
  >
    <div
      className="emergencyNumberDescription"
    >
      <p />
    </div>
  </div>
  <div
    className="emergencyNumberContainer"
  >
    <div
      className="emergencyNumber"
    >
      <div>
        <span>
          : 
        </span>
        <a
          href="tel:undefined"
        />
      </div>
    </div>
    <div
      className="emergencyNumber"
    >
      <div>
        <span>
          : 
        </span>
        <a
          href="tel:undefined"
        />
      </div>
    </div>
    <div
      className="emergencyNumber"
    >
      <div>
        <span>
          : 
        </span>
        <a
          href="tel:undefined"
        />
      </div>
    </div>
    <div
      className="emergencyNumberEditButtons"
    >
      <div
        className="emergencyNumberEditIcon"
        role="presentation"
      >
        <span
          className="icon icon--pencil"
        >
          <svg
            viewBox="0 0 50 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
            />
          </svg>
        </span>
      </div>
      <div
        className="emergencyNumberIcon"
        role="presentation"
      >
        <span
          className="icon icon--archive"
        >
          <svg
            viewBox="0 0 50 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M37.2 18h-25c-.8 0-1.5-.7-1.5-1.5s.7-1.5 1.5-1.5h24.9c.8 0 1.5.7 1.5 1.5S38 18 37.2 18zm-2.7-6.4H14.9c-.8 0-1.5-.7-1.5-1.5s.7-1.5 1.5-1.5h19.6c.8 0 1.5.7 1.5 1.5s-.7 1.5-1.5 1.5z"
            />
            <path
              d="M41.9 3.7a3.4 3.4 0 0 0-3.2-2.1H11.3c-1.4 0-2.6.8-3.2 2.1L0 23.4v23.8c0 1.1.9 2 2 2h46a2 2 0 0 0 2-2V23.4L41.9 3.7zM10.8 4.9c.1-.2.3-.3.4-.3h27.5c.2 0 .4.1.4.3l7.6 18.5H3.2l7.6-18.5zM36 35c-.9 1.1-4.4 4.7-10 5h-.8a15 15 0 0 1-10.9-5.1c-.5-.6-.4-1.4.1-1.9.6-.5 1.4-.4 1.9.1a12 12 0 0 0 9.4 4.1c4.5-.3 7.3-3.2 8.1-4.1.5-.6 1.4-.6 1.9-.2.7.7.8 1.5.3 2.1z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`EmergencyNumberStatic Component renders phoneNumberTwo if it exists 1`] = `
<div
  className="emergencyNumberItem"
>
  <div
    className="emergencyTitle"
  >
    <div
      className="emergencyNumberDescription"
    >
      <p />
    </div>
  </div>
  <div
    className="emergencyNumberContainer"
  >
    <div
      className="emergencyNumber"
    >
      <div>
        <span>
          : 
        </span>
        <a
          href="tel:undefined"
        />
      </div>
    </div>
    <div
      className="emergencyNumber"
    >
      <div>
        <span>
          : 
        </span>
        <a
          href="tel:undefined"
        />
      </div>
    </div>
    <div
      className="emergencyNumberEditButtons"
    >
      <div
        className="emergencyNumberEditIcon"
        role="presentation"
      >
        <span
          className="icon icon--pencil"
        >
          <svg
            viewBox="0 0 50 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M42.9 17.2L12.4 47.7c-.6.6-1.4 1-2.3 1.1l-9 .9c-.4 0-.8-.3-.7-.7l.9-9c.1-.9.5-1.7 1.1-2.3L32.9 7.2l10 10zm1.5-1.5l-10-10 4.4-4.4c.5-.5 1.4-.5 1.9 0l8.1 8.1c.5.5.5 1.4 0 1.9l-4.4 4.4z"
            />
          </svg>
        </span>
      </div>
      <div
        className="emergencyNumberIcon"
        role="presentation"
      >
        <span
          className="icon icon--archive"
        >
          <svg
            viewBox="0 0 50 50"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M37.2 18h-25c-.8 0-1.5-.7-1.5-1.5s.7-1.5 1.5-1.5h24.9c.8 0 1.5.7 1.5 1.5S38 18 37.2 18zm-2.7-6.4H14.9c-.8 0-1.5-.7-1.5-1.5s.7-1.5 1.5-1.5h19.6c.8 0 1.5.7 1.5 1.5s-.7 1.5-1.5 1.5z"
            />
            <path
              d="M41.9 3.7a3.4 3.4 0 0 0-3.2-2.1H11.3c-1.4 0-2.6.8-3.2 2.1L0 23.4v23.8c0 1.1.9 2 2 2h46a2 2 0 0 0 2-2V23.4L41.9 3.7zM10.8 4.9c.1-.2.3-.3.4-.3h27.5c.2 0 .4.1.4.3l7.6 18.5H3.2l7.6-18.5zM36 35c-.9 1.1-4.4 4.7-10 5h-.8a15 15 0 0 1-10.9-5.1c-.5-.6-.4-1.4.1-1.9.6-.5 1.4-.4 1.9.1a12 12 0 0 0 9.4 4.1c4.5-.3 7.3-3.2 8.1-4.1.5-.6 1.4-.6 1.9-.2.7.7.8 1.5.3 2.1z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
